client/src/App.jsx

```diff
@@ -0,0 +1,33 @@
+import Home from "./Pages/Home"
+import { BrowserRouter, Route, Routes,Navigate} from 'react-router-dom'
+import PostBlog from "./Pages/PostBlog"
+import SearchBlog from "./Pages/SearchBlog"
+import BlogCategoryPage from "./Pages/blogCategoryPage"
+import i18 from 'i18next'
+import Login from "./Pages/Login"
+import Register from "./Pages/Register"
+
+function App() {
+
+  // const {lng} = useParams();
```

Comment:
if this is commented out how are paths below in the file resolved w.r.t `lng`

Reply:
Beacuse we are accessing the language by the user's `browser sessions`.
```bash
{/* Fallback route to redirect to default language 'en' */}
        <Route path="/" element={<Navigate to={`/${i18.language}`}/>} />

        <Route path={"/:lng"} element={<Home/>}/>
        <Route path={"/:lng/login"} element={<Login/>}/>
        <Route path={"/:lng/register"} element={<Register/>}/>
        <Route path="/:lng/post-blog" element={<PostBlog/>}/>
        <Route path="/:lng/search-blog" element={<SearchBlog/>}/>
        <Route path="/:lng/blog/:category" element={<BlogCategoryPage/>} />
```
By default it will be set on `Eng` but if the user will selects `Urdu` the `lang:ur` will be stored in his `cookies` and then will be added in `param's` which will be later handled on other pages.